### PR TITLE
PD-616: Adds closeOnBackdropClick prop

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Addresses: Cannot prevent closing modal when clicking backdrop by adding `closeOnBackdropClick` prop

🎟 _Jira ticket:_ [PD-616](https://jira.mongodb.org/browse/PD-616)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: packages/modal/src/Modal.spec.tsx [hunk 1]
```diff
@@ -58,6 +58,12 @@ describe('packages/modal', () => {
       window.dispatchEvent(new MouseEvent('click', { bubbles: true }));
       expect(container.innerHTML).toBe('');
     });
+
+    test('when "closeOnBackdropClick" is false', () => {
+      renderModal({ closeOnBackdropClick: false, open: true });
+      window.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+      expect(screen.getByText(modalContent)).toBeInTheDocument();
```
Comment (DesignerDave) [inline packages/modal/src/Modal.spec.tsx:65]: I still don't like the pattern of selecting by text and other attributes, and would prefer a test-id. Technically, this only gets a child of the modal, and doesn't test that the whole modal is still in the document.

Comment (brookescarlett) [reply packages/modal/src/Modal.spec.tsx:65]: Happy to switch this, just trying to follow the documented preferred patterns but if you feel strongly, let me know.

File: packages/modal/src/Modal.spec.tsx [hunk 2]
```diff
@@ -58,6 +58,12 @@ describe('packages/modal', () => {
       window.dispatchEvent(new MouseEvent('click', { bubbles: true }));
       expect(container.innerHTML).toBe('');
     });
+
+    test('when "closeOnBackdropClick" is false', () => {
```
[the remark this answers is not in the document]
Comment (brookescarlett) [reply packages/modal/src/Modal.spec.tsx:62]: Added!

File: packages/modal/src/Modal.spec.tsx (hunk 1 above)
Comment (DesignerDave) [reply packages/modal/src/Modal.spec.tsx:65]: I think I feel strongly enough about this – especially given that it doesn't technically test this correctly (as noted above).

Comment (brookescarlett) [reply packages/modal/src/Modal.spec.tsx]: Fixed @DesignerDave!

**Pull request merged**